src/auth/auth.module.ts

```diff
@@ -11,4 +15,41 @@ import { ContactModule } from '../contact/contact.module';
   providers: [AuthService],
   exports: [AuthService],
 })
-export class AuthModule {}
+export class AuthModule {
+  SKIP_AUTH = false;
+  CRM_IMPOSTER_EMAIL = 'labs_dl@planning.nyc.gov';
+
+  // this must match what is in the eponymous env variable
+  NYCID_CONSOLE_PASSWORD = 'test';
+
+  constructor(private readonly config: ConfigService) {
+    this.SKIP_AUTH = this.config.get('SKIP_AUTH');
+    this.CRM_IMPOSTER_EMAIL = this.config.get('CRM_IMPOSTER_EMAIL');
+    this.NYCID_CONSOLE_PASSWORD = this.config.get('NYCID_CONSOLE_PASSWORD');
+  }
+
+  onApplicationBootstrap() {
+    if (this.SKIP_AUTH) {
+      this._printCookieInConsole();
+    }
+  }
+
+  _printCookieInConsole() {
+    const { NYCID_CONSOLE_PASSWORD, CRM_IMPOSTER_EMAIL } = this;
+    const mockJWT = jwt.sign({
```

Comment:
Can you explain more what is going on here? Usually `jwt.sign` is passed a CRM_IMPOSTER_ID. 
How does a token work for future requests if it is signed using an email?

Reply:
Hmm what's going on here is automating how authentication works in this app. `jwt.sign` is not necessarily or even usually passed an imposter ID. In fact, only when imposter ID is enabled is it ever passed that. Usually it is passed a real contactid, which is inferred from an e-mail address that's provided by NYC.ID's JWT. So, there are _two_ kinds of JWTs in play in this application:
 - NYC.ID JWT
 - Our JWT

In order to login to this application, you must first be authenticated through NYC.ID. Once authenticated, NYC.ID redirects you back to this application with a _JWT that includes your e-mail address_. That is what you are seeing on line 39: we are generating a _mock_ NYC.ID JWT and _signing_ it with the `NYCID_CONSOLE_PASSWORD` _as if_ you were authenticated via NYC.ID.

So `_printCookieInConsole` recreates the NYC.ID authentication step for use in local development. The intent is to enable local debugging that is closer to _how authentication actually works_.

That's why I remove the references to `SKIP_AUTH` in the `auth` service: instead of sprinkling development-only "exception" behavior throughout the class (thereby changing its behavior), we fully use the public interface of login.

Happy to add comments.

Reply:
Once the cookie is printed in the terminal, you can copy that cookie and paste it into headers of requests you're making. If you're looking at the API and want it to persist, you can use the Chrome Inspector to add the token via the Application tab:

<img width="830" alt="Screen Shot 2020-01-24 at 12 34 59 PM" src="https://user-images.githubusercontent.com/5004319/73090233-f63de980-3ea5-11ea-9824-de993f033b33.png">

Reply:
Ah okay thanks, I get it now. Requesting some documentation and code naming changes